### PR TITLE
fix(Subs): Allow to select a part of a single line

### PR DESCRIPTION
**Pull request by peterfication**

### CODE REVIEW

**Comment by peterfication** [comment]
@johmsalas another thing I need to fix before the 1.0.0 release. I came across this while writing a short blog post about my last fix.

File: lua/textcase/plugin/plugin.lua [hunk 1]
```diff
@@ -150,8 +150,12 @@ function M.incremental_substitute(opts, preview_ns, preview_buf)
...
 
   -- Equivalent to the method TextCaseSubstituteLauncher (start.vim) computing
-  -- if it is a multiline selection to use either normal or visual mode
-  local mode = (opts.range < 2 or opts.line1 == opts.line2) and "n" or visual_mode
+  -- if it is a multiline selection to use either normal or visual mode.
+  --
+  -- If the count is -1, it means that the command was NOT called from a visual
+  -- mode. We only care about visual modes or normal mode, so we can set the
+  -- mode to "n" in that case.
+  local mode = opts.count == -1 and "n" or visual_mode
```
Comment (johmsalas) [inline lua/textcase/plugin/plugin.lua:158]: At this point I'm going to propose a vim-mode-decoder plugin xD jk

**Review by johmsalas — APPROVED** [review]
Good catch. Will keep an eye your blog

File: lua/textcase/plugin/plugin.lua (hunk 1 above)
Comment (peterfication) [reply lua/textcase/plugin/plugin.lua:158]: :D I already made a local TODO comment in the code there `Extract mode retrieval into utils function`

**Pull request merged**